src/Widget/BaseWidget.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Widget;
+
+use Bolt\Snippet\Target;
+use Bolt\Snippet\Zone;
+use Cocur\Slugify\Slugify;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Twig\Environment;
+
+class BaseWidget
+{
+    protected $name = 'Nameless widget';
+    protected $type = 'widget';
+    protected $target = Target::NOWHERE;
+    protected $zone = Zone::EVERYWHERE;
+    protected $priority = 0;
+    protected $context = [];
+
+    /** @var Environment */
+    protected $twig;
+
+    /** @var string */
+    protected $template;
+
+    /** @var Request */
+    protected $request;
+
+    /** @var Response */
+    protected $response;
+
+    public function setName(string $name): self
```

Comment:
check if all those getters and setters are even used

Reply:
They are used where I instantiate the `SnippetWidget`, but even they aren't: This is a piece of code that will be used by others to extend Bolt, so it'll be usefull to have them.

Reply:
Not when there will be an interface. Values can be set during construction.